server/pulp/server/db/model/__init__.py

```diff
@@ -847,8 +859,31 @@ class LazyCatalogEntry(AutoRetryDocument):
     url = StringField(required=True)
     checksum = StringField()
     checksum_algorithm = StringField(regex=ALG_REGEX)
+    revision = IntField(default=0)
     data = DictField()
 
+    def save_revision(self):
+        """
+        Add the entry using the next revision number.
+        Previous revisions are deleted.
+        """
+        revisions = {0}
```

Comment:
I believe this needs to be ``set([0])`` since set literals are Python 2.7+[0].

[0] https://docs.python.org/3/whatsnew/2.7.html#python-3-1-features